modules/openxoutstreamBidAdapter.js

```diff
@@ -0,0 +1,216 @@
+    nocache: new Date().getTime()
+  };
+
+  if (utils.deepAccess(bidderRequest, 'gdprConsent')) {
+    let gdprConsentConfig = bidderRequest.gdprConsent;
+
+    if (gdprConsentConfig.consentString !== undefined) {
+      defaultParams.gdpr_consent = gdprConsentConfig.consentString;
+    }
+
+    if (gdprConsentConfig.gdprApplies !== undefined) {
+      defaultParams.gdpr = gdprConsentConfig.gdprApplies ? 1 : 0;
+    }
+
+    if (config.getConfig('consentManagement.cmpApi') === 'iab') {
+      defaultParams.x_gdpr_f = 1;
+    }
+  }
+
+  return defaultParams;
+}
+
+function buildOXBannerRequest(bids, bidderRequest) {
+  let queryParams = buildCommonQueryParamsFromBids(bids, bidderRequest);
+  queryParams.aus = utils._map(bids, bid => utils.parseSizesInput(bid.sizes).join(',')).join('|');
+
+  if (bids.some(bid => bid.params.doNotTrack)) {
+    queryParams.ns = 1;
+  }
+
+  if (bids.some(bid => bid.params.coppa)) {
+    queryParams.tfcd = 1;
+  }
+
+  let url = `https://${bids[0].params.delDomain}/v/1.0/avjp`
+  return {
+    method: 'GET',
+    url: url,
+    data: queryParams,
+    payload: {'bids': bids}
+  };
+}
+
+function handleVastResponse(response, serverResponse) {
+  const body = response.body
+  let bidResponses = [];
+  if (response !== undefined && body.vastUrl !== '' && body.pub_rev && body.pub_rev > 0) {
+    const openHtmltag = '<html><head><meta name="viewport" content="width=device-width, initial-scale=1.0"></head><body>';
+    const closeHtmlTag = '</body></html>';
+    const sdkScript = createSdkScript().outerHTML;
+    const placementDiv = createPlacementDiv();
+    placementDiv.dataset.pID = PUBLISHER_ID;
+    const placementDivString = placementDiv.outerHTML;
+    const adResponse = getTemplateAdResponse(body.vastUrl);
+    const adResponseString = JSON.stringify(adResponse);
+    const ymAdsScript = '<script type="text/javascript"> window.__ymAds =' + adResponseString + '</script>';
+
+    let bidResponse = {};
+    bidResponse.requestId = serverResponse.bids[0].bidId;
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.netRevenue = NET_REVENUE;
+    bidResponse.currency = CURRENCY;
+    bidResponse.cpm = Number(body.pub_rev) / 1000;
+    bidResponse.creativeId = body.adid;
+    bidResponse.height = body.height;
+    bidResponse.width = body.width;
+    bidResponse.vastUrl = body.vastUrl;
+    bidResponse.ttl = TIME_TO_LIVE;
+    bidResponse.mediaType = BANNER;
+    bidResponse.ad = openHtmltag + placementDivString + ymAdsScript + sdkScript + closeHtmlTag;
+
+    bidResponses.push(bidResponse);
+  }
+  return bidResponses;
+}
+registerBidder(spec);
+
+// HELPER FUNCTIONS
+function createSdkScript() {
+  const script = document.createElement('script');
+  script.innerHTML = YM_SCRIPT;
+  return script;
+}
+function createPlacementDiv() {
+  const div = document.createElement('div');
+  div.id = `ym_${PLACEMENT_ID}`;
+  div.classList.add('ym');
+  div.dataset.lfid = CR_ID;
```

Comment:
This has to be 'div.dataset.lfId' with a capital 'i' in the 'lfId', otherwise sdk won't parse it properly